validation-test/BuildSystem/LTO/add_host_variant_c_compile_link_flags_characterization_lto.test

```diff
@@ -0,0 +1,50 @@
+# REQUIRES: standalone_build
+
+# RUN: %empty-directory(%t)
+# RUN: mkdir -p %t/build
+# RUN: mkdir -p %t/source
+# RUN: split-file %s %t/source
+# RUN: %{cmake-generate} -DSWIFT_TOOLS_ENABLE_LTO="thin" -DCMAKE_MODULE_PATH=%swift_src_root/cmake/modules -S %t/source -B %t/build | %FileCheck --check-prefix=CHECK-THIN %s
+
+# CHECK-THIN: COMPILE_OPTIONS
+# CHECK-THIN-SAME: -flto=thin
+# CHECK-THIN: LINK_OPTIONS
+# CHECK-THIN-SAME: -flto=thin
+
+# RUN: %empty-directory(%t/build)
+# RUN: %{cmake-generate} -DSWIFT_TOOLS_ENABLE_LTO="full" -DCMAKE_MODULE_PATH=%swift_src_root/cmake/modules -S %t/source -B %t/build | %FileCheck --check-prefix=CHECK-FULL %s
+
+# CHECK-FULL: COMPILE_OPTIONS
+# CHECK-FULL-SAME: -flto=full
+# CHECK-FULL: LINK_OPTIONS
+# CHECK-FULL-SAME: -flto=full
+
+# RUN: %empty-directory(%t/build)
+# RUN: %{cmake-generate} -DSWIFT_TOOLS_ENABLE_LTO="" -DCMAKE_MODULE_PATH=%swift_src_root/cmake/modules -S %t/source -B %t/build | %FileCheck %s
```

Comment:
Feedback from Michael: it is more valuable to generate the build system with Ninja and use Ninja itself for the assertion -- this way the test would be on the generated build system (the one we will be using), and not about the internal representation that CMake uses to generate the build system.